### PR TITLE
Add a param to enable/disable nam transfers

### LINKED ISSUE #2842: Parameter to disable NAM transfers
We should add a governance-controlled parameter that enables/disables specifically native token (NAM) transfers. This can be accomplished by reading this parameter in the multitoken VP, and rejecting unless it is 1 (enabled).

### PR DESCRIPTION
**Pull request by yito88**
## Describe your changes
closes #2842 

## Indicate on which release or other PRs this topic is based on
`v0.31.9`

## Checklist before merging to `draft`
- [x] I have added a changelog
- [x] Git history is in acceptable state

### CODE REVIEW

**Review by grarco — DISMISSED** [review]
There's an issue with `make check-crates` but apart from that looks good to me, thank you!

File: crates/parameters/src/lib.rs [hunk 1]
```diff
@@ -474,3 +488,33 @@ where
 pub fn native_erc20_key() -> Key {
     storage::get_native_erc20_key_at_addr(ADDRESS)
 }
+
+/// Initialize parameters to the storage for testing
+#[cfg(any(test, feature = "testing"))]
```
Comment (grarco) [inline crates/parameters/src/lib.rs:493]: I believe `test` is not propagated to the dependency files and so we end up not finding this function

**Review by cwgoes — DISMISSED** [review]
lgtm

**Comment by Fraccaman** [comment]
will this prevent bonding? I think we need to be able to transfer nam to the PoS account if we need to bond.

**Comment by cwgoes** [comment]
> will this prevent bonding? I think we need to be able to transfer nam to the PoS account if we need to bond.

Ah, good point - in that case we should allow transfers to (but not from) the PoS account.

**Comment by brentstone** [comment]
> > will this prevent bonding? I think we need to be able to transfer nam to the PoS account if we need to bond.
> 
> Ah, good point - in that case we should allow transfers to (but not from) the PoS account.

More particularly, perhaps we should allow transfers between `Internal` addresses and `Implicit / Established` addresses, since we need the same functionality with Governance and then the MASP before transfers are enabled (and maybe others I'm not considering)

**Comment by cwgoes** [comment]
> > > will this prevent bonding? I think we need to be able to transfer nam to the PoS account if we need to bond.
> > 
> > 
> > Ah, good point - in that case we should allow transfers to (but not from) the PoS account.
> 
> More particularly, perhaps we should allow transfers between `Internal` addresses and `Implicit / Established` addresses, since we need the same functionality with Governance and then the MASP before transfers are enabled (and maybe others I'm not considering)

I don't quite follow - why would we need to transfer NAM (specifically) to and from the MASP or governance, before NAM transfers are enabled?

**Comment by cwgoes** [comment]
> Gov proposals require that the author send NAM to the gov address, which is kept if the proposal fails I believe (and refunded if passed). And if you want to shield NAM, don't we send it to the MASP address?

Governance proposal deposits are OK. We do not want to allow shielding NAM until transfers are enabled.

**Pull request merged**